content/docs/specs/remote_write_spec_2_0.md

```diff
@@ -141,17 +141,17 @@ The following subsections specify Sender and Receiver semantics around headers a
-X-Prometheus-Remote-Write-Accepted-Samples <integer; count of all written Samples from this request>
-X-Prometheus-Remote-Write-Accepted-Histograms <integer; count of all written Histogram samples from this request>
-X-Prometheus-Remote-Write-Accepted-Exemplars <integer; count of all written Exemplars from this request>
+* `X-Prometheus-Remote-Write-Accepted-Samples <integer; count of all accepted Samples from this request>`
+* `X-Prometheus-Remote-Write-Accepted-Histograms <integer; count of all accepted Histogram samples from this request>`
+* `X-Prometheus-Remote-Write-Accepted-Exemplars <integer; count of all accepted Exemplars from this request>`
 
-Senders MAY use those headers for client instrumentation and communication durability checks, e.g. to verify no miscommunication or implementation bug exists between the Sender and Receiver. Specifically, this guards against broken Sender or Receiver implementations with a buggy or missing content-type checks and accidental decoding of the `io.prometheus.write.v2.Request` payload with `prometheus.WriteRequest` schema (no error on decoding and an empty result).
+Senders MAY use those headers to confirm how many Samples, Histograms and Exemplars were accepted or successfully ingested by the Receiver, which is especially useful in [Partial Write](#partial-write) failure situations. Senders MAY also use those headers for client instrumentation and communication durability checks, e.g. to verify no miscommunication or implementation bug exists between the Sender and Receiver. Specifically, this guards against broken Sender or Receiver implementations with a buggy or missing content-type checks and accidental decoding of the `io.prometheus.write.v2.Request` payload with `prometheus.WriteRequest` schema (no error on decoding and an empty result).
 
-For example, Senders MAY assume [415 HTTP Unsupported Media Type](https://www.rfc-editor.org/rfc/rfc9110.html#name-415-unsupported-media-type) status code, when `io.prometheus.write.v2.Request` request with samples, results in 2xx HTTP status code, but no response headers from the Receiver (a common issue for the 1.0 Reciever without content-type header check).
+Senders MUST treat the omission of any `X-Prometheus-Remote-Write-Accepted-*` header as either no Samples, Histograms or Exemplars were accepted by the Receiver or the Receiver only supports Remote Write 1.0 protocol and `prometheus.WriteRequest` Protobuf Message. For example, Senders MAY assume [415 HTTP Unsupported Media Type](https://www.rfc-editor.org/rfc/rfc9110.html#name-415-unsupported-media-type) status code, when the `io.prometheus.write.v2.Request` request with samples, results in 2xx HTTP status code, but no `X-Prometheus-Remote-Write-Accepted-*` response headers from the Receiver (a common issue for the 1.0 Receiver without content-type header check). Senders MUST NOT assume the same for `prometheus.WriteRequest` (for compatibility reasons) or when receiving zero counts (Receiver might )
```

Comment:
this would be simpler if we didn't allow for omitting the header if the value was 0

Reply:
Yes and no - we discussed this heavily with @cstyan  in https://cloud-native.slack.com/archives/C06B8KJUU2Y/p1720628447889389 will try to put the consensu on the PR. 🤗